### PR TITLE
tests: make TestBasicPayouts less flaky

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Similarly to #5995 sync fixture and test client.

## Test Plan

Flaky test fix

### CODE REVIEW

**Review by gmalouf — DISMISSED** [review]

**Review by jasonpaulos — DISMISSED** [review]

File: test/e2e-go/features/incentives/payouts_test.go [hunk 1]
```diff
@@ -344,6 +349,11 @@ func rekeyreg(f *fixtures.RestClientFixture, a *require.Assertions, client libgo
 	a.NoError(err)
 	txn, err := f.WaitForConfirmedTxn(uint64(reReg.LastValid), onlineTxID)
 	a.NoError(err)
+	// sync up with the network
+	status, err := f.LibGoalClient.Status()
+	a.NoError(err)
+	_, err = client.WaitForRound(status.LastRound)
+	a.NoError(err)
```
Comment (jannotti) [inline test/e2e-go/features/incentives/payouts_test.go:356]:
```suggestion
	_, err = client.WaitForRound(*txn.ConfirmedRound)
	a.NoError(err)
```

File: test/e2e-go/features/incentives/payouts_test.go [hunk 2]
```diff
@@ -81,14 +81,19 @@ func TestBasicPayouts(t *testing.T) {
 	a.NoError(err)
 	burn, err := fixture.WaitForConfirmedTxn(uint64(txn.LastValid), txn.ID().String())
 	a.NoError(err)
+	// sync up with the network
+	status, err := fixture.LibGoalClient.Status()
+	a.NoError(err)
+	_, err = c01.WaitForRound(status.LastRound)
+	a.NoError(err)
```
Comment (jannotti) [inline test/e2e-go/features/incentives/payouts_test.go:88]:
```suggestion
	_, err = c01.WaitForRound(*burn.ConfirmedRound)
	a.NoError(err)
```

File: test/e2e-go/features/incentives/payouts_test.go [hunk 3]
```diff
@@ -81,14 +81,19 @@ func TestBasicPayouts(t *testing.T) {
...
 	data01, err = c01.AccountData(account01.Address)
 	a.NoError(err)
 
 	// Go 31 rounds after the burn happened. During this time, incentive
 	// eligibility is not in effect yet, so regardless of who proposes, they
 	// won't earn anything.
 	client := fixture.LibGoalClient
-	status, err := client.Status()
+	status, err = client.Status()
```
Comment (algorandskiy) [inline test/e2e-go/features/incentives/payouts_test.go:96]:
```suggestion
	status, err := client.Status()
```

**Comment by algorandskiy** [comment]
applied suggestions

**Review by jannotti — APPROVED** [review]

**Review by gmalouf — APPROVED** [review]

**Pull request merged**